alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+    features_list
+        A list of features for which to plot the partial dependence curves or ``'all'`` for all features.
+        Can be an integers denoting feature index denoting entries in `exp.feature_names`. Defaults to ``'all'``.
+    target_idx
+        Target index for which to plot the partial dependence (PD) curves. Can be a mix of integers denoting target
+        index or strings denoting entries in `exp.meta['target_names']`.
+    n_cols
+        Number of columns to organize the resulting plot into.
+    n_ice
+        Number of ICE plots to be displayed. Can be
+
+         - a string taking the value ``'all'`` to display the ICE curves for every instance in the reference dataset.
+
+         - an integer for which `n_ice` instances from the reference dataset will be sampled uniformly at random to
+           display their ICE curves.
+
+         - a list of integers, where each integer represents an index of an instance in the reference dataset to
+           display their ICE curves.
+
+    centered
+        Boolean flag to center the individual conditional expectation (ICE) curves.
+    levels
+        Number of levels in the contour plot.
+    ax
+        A `matplotlib` axes object or a `numpy` array of `matplotlib` axes to plot on.
+    sharey
+        A parameter specifying whether the y-axis of the PD and ICE curves should be on the same scale
+        for several features. Possible values are: ``'all'`` | ``'row'`` | ``None``.
+    pd_num_kw
+        Keyword arguments passed to the `matplotlib plot` function when plotting the PD for a numerical feature.
+    ice_num_kw
+        Keyword arguments passed to the `matplotlib plot` function when plotting the ICE for a numerical feature.
+    pd_cat_kw
+        Keyword arguments passed to the `matplotlib plot` function when plotting the PD for a categorical feature.
+    ice_cat_kw
+        Keyword arguments passed to the `matplotlib plot` function when plotting the ICE for a categorical feature.
+    pd_num_num_kw
+        Keyword arguments passed to the `matplotlib contourf` function when plotting the PD for two numerical features.
+    pd_num_cat_kw
+        Keyword arguments passed to the `matplotlib plot` function when plotting the PD for a numerical and a
+        categorical feature.
+    pd_cat_cat_kw
+        Keyword arguments passed to the `seaborn heatmap` functon when plotting the PD for two categorical features.
+    fig_kw
+        Keyword arguments passed to the `fig.set` function.
+    seed
+        Seed to be used for ICE sampling.
+
+    Returns
+    -------
+    An array of `matplotlib` axes with the resulting partial dependence plots.
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib.gridspec import GridSpec
+
+    default_fig_kw = {'tight_layout': 'tight'}
+    if fig_kw is None:
+        fig_kw = {}
+    fig_kw = {**default_fig_kw, **fig_kw}
+
+    if features_list == 'all':
+        features_list = range(0, len(exp.feature_names))
```

Comment:
To clarify - once the explanation is returned then `exp.feature_names` is actually a combination of the original `feature_names` and some 2-way feature name combinations (if any 2-way PDs are requested)?

Reply:
yes. For example `exp.feature_names = ['temp', ('temp', 'windspeed')]`.